### PR TITLE
finished basic menu component functionality and design

### PR DESCRIPTION
**Pull request by cminho921**
changed schema design, finished the menu component part and added CSS style file

### CODE REVIEW

File: client/dist/menu.css [hunk 1]
```diff
@@ -0,0 +1,389 @@
+body {
+  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
+  /* background-color: #f1f2f4; */
```
Comment (LittleJennie) [inline client/dist/menu.css]: please uncomment or cleanup commented codes if they are not necessary css

File: client/dist/menu.css [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+  left:0;bottom:0;
+  width:100%;
+  background-image:-webkit-gradient(linear,left bottom, left top, from(#fff), to(rgba(255, 255, 255, 0)));
+  background-image: linear-gradient(to top, #fff, rgba(255, 255, 255, 0));
+  background-color: rgba(255, 255, 255, .2)
+}
+
+.menu-price__1KiCimAr {
+  float: right;
+
```
Comment (LittleJennie) [inline client/dist/menu.css]: remove unnecessary line

File: client/dist/menu.css [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+.menu-price__1KiCimAr:before {
+  content: '$';
+}
+
+.menu-price__1KiCimAr:after {
+  content: '.00';
+} 
+
+
```
Comment (LittleJennie) [inline client/dist/menu.css]: remove unnecessary line

File: client/dist/menu.css [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+.container__1I7-tNMk {
+  position: relative
+}
+
+.sectionHeader__ByIiO8gW {
+  font-size: 24px;
+  font-weight: 700;
+  line-height: 32px;
+  color: #2d333f;
+  /* border-bottom: 1px solid #d8d9db; */
```
Comment (LittleJennie) [inline client/dist/menu.css]: remove unnecessary line

File: client/src/components/App.jsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import axios from 'axios';
 // eslint-disable-next-line import/extensions
```
Comment (LittleJennie) [inline client/src/components/App.jsx]: looks like all the eslint-disable-next-line appears couple times here and all commented out?

File: client/src/components/App.jsx [hunk 6]
```diff
@@ -12,39 +12,90 @@ class App extends React.Component {
...
-        this.setState({ restaurantData: restaurantsData[0] });
+        this.setState({
+          restaurantData: restaurantsData[0],
+          currentSubMenu: restaurantsData[0].menu_category[0].submenu_name,
+        });
       });
   }
 
+  handleMenuCategoryClick(e) {
+    console.log(e.target.id);
```
Comment (LittleJennie) [inline client/src/components/App.jsx]: remove unnecessary console log

File: client/src/components/App.jsx [hunk 7]
```diff
@@ -12,39 +12,90 @@ class App extends React.Component {
...
+  handleCollapseButtonClick() {
+    const { currentCollapseToggle } = this.state;
+    const newToggle = !currentCollapseToggle;
+    this.setState({ currentCollapseToggle: newToggle });
+  }
+
   render() {
     const { restaurantData } = this.state;
+    const { currentSubMenu } = this.state;
+    const { currentCollapseToggle } = this.state;
```
Comment (LittleJennie) [inline client/src/components/App.jsx]: line 47 to 49, please combined to one line. 
```suggestion
    const { restaurantData , currentSubMenu , currentCollapseToggle } = this.state;
```

File: client/src/components/App.jsx [hunk 8]
```diff
@@ -12,39 +12,90 @@ class App extends React.Component {
...
+
+  handleCollapseButtonClick() {
+    const { currentCollapseToggle } = this.state;
+    const newToggle = !currentCollapseToggle;
+    this.setState({ currentCollapseToggle: newToggle });
+  }
+
   render() {
     const { restaurantData } = this.state;
+    const { currentSubMenu } = this.state;
```
Comment (LittleJennie) [inline client/src/components/App.jsx]:
```suggestion
```

File: client/src/components/App.jsx [hunk 9]
```diff
@@ -12,39 +12,90 @@ class App extends React.Component {
...
+  }
+
+  handleCollapseButtonClick() {
+    const { currentCollapseToggle } = this.state;
+    const newToggle = !currentCollapseToggle;
+    this.setState({ currentCollapseToggle: newToggle });
+  }
+
   render() {
     const { restaurantData } = this.state;
```
Comment (LittleJennie) [inline client/src/components/App.jsx]:
```suggestion
```

File: client/src/components/menuCategoryButtons.jsx [hunk 10]
```diff
@@ -1,9 +1,13 @@
...
-    <button className="button-menu-header" type="button">{ category }</button>
-  ))
+  <div>
+    { props.restaurants.map(category => (
+      category.submenu_name === props.currentSubMenu
+        ? <button className="button-menu-link-active__3KrRGgFS button-menu-link__1j5RFIKk secondary__37FrTgAl base__1GdpdUto" type="button" id={category.submenu_name} onClick={props.handleMenuCategoryClick}>{category.submenu_name}</button>
+        : <button className="button-menu-link__1j5RFIKk secondary__37FrTgAl base__1GdpdUto" type="button" id={category.submenu_name} onClick={props.handleMenuCategoryClick}>{category.submenu_name}</button>
+    ))}
+  </div>
 );
```
Comment (LittleJennie) [inline client/src/components/menuCategoryButtons.jsx]: I think you might need props validation here?

File: client/src/components/menuCategoryButtons.jsx [hunk 11]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
 
 const MenuCategoryButtons = props => (
-  props.restaurants.map(category => (
-    <button className="button-menu-header" type="button">{ category }</button>
-  ))
+  <div>
+    { props.restaurants.map(category => (
+      category.submenu_name === props.currentSubMenu
+        ? <button className="button-menu-link-active__3KrRGgFS button-menu-link__1j5RFIKk secondary__37FrTgAl base__1GdpdUto" type="button" id={category.submenu_name} onClick={props.handleMenuCategoryClick}>{category.submenu_name}</button>
```
Comment (LittleJennie) [inline client/src/components/menuCategoryButtons.jsx]: For styling, it's better to put them in separate line
```suggestion
        ? <button 
            className="button-menu-link-active__3KrRGgFS button-menu-link__1j5RFIKk secondary__37FrTgAl base__1GdpdUto"
            type="button" id={category.submenu_name} 
            onClick={handleMenuCategoryClick}
         >
           {category.submenu_name}
         </button>
```

File: client/src/components/menuCategoryButtons.jsx [hunk 12]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
 
 const MenuCategoryButtons = props => (
```
Comment (LittleJennie) [inline client/src/components/menuCategoryButtons.jsx]: Destructure props when you invoke the component function.
```suggestion
const MenuCategoryButtons =({
  restaurants,
  currentSubMenu,
  handleMenuCategoryClick,
})=> (
```

File: client/src/components/menuCategoryButtons.jsx [hunk 13]
```diff
@@ -1,9 +1,13 @@
...
-  props.restaurants.map(category => (
-    <button className="button-menu-header" type="button">{ category }</button>
-  ))
+  <div>
+    { props.restaurants.map(category => (
+      category.submenu_name === props.currentSubMenu
+        ? <button className="button-menu-link-active__3KrRGgFS button-menu-link__1j5RFIKk secondary__37FrTgAl base__1GdpdUto" type="button" id={category.submenu_name} onClick={props.handleMenuCategoryClick}>{category.submenu_name}</button>
+        : <button className="button-menu-link__1j5RFIKk secondary__37FrTgAl base__1GdpdUto" type="button" id={category.submenu_name} onClick={props.handleMenuCategoryClick}>{category.submenu_name}</button>
```
Comment (LittleJennie) [inline client/src/components/menuCategoryButtons.jsx]: Same as line 7, for styling, it's better to put them in separate line.

File: client/src/components/menuEntry.jsx [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+  props.entries.map(entry => (
+    <div className="menu-item__2ZxJOnTY">
+      <div className="menu-price__1KiCimAr">{entry.menu_price}</div>
+      <div className="entry-item-name">{entry.menu_name}</div>
+      <div className="menu-description__2HXkC4oE">{entry.menu_description}</div>
+    </div>
+  ))
+);
+
+export default MenuEntry;
```
Comment (LittleJennie) [inline client/src/components/menuEntry.jsx]: need extra line

File: client/src/components/menuEntry.jsx [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const MenuEntry = props => (
```
Comment (LittleJennie) [inline client/src/components/menuEntry.jsx]:
```suggestion
const MenuEntry =({ entries }) => (
```

File: client/src/components/menuCategoryButtons.jsx [hunk 16]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
 
 const MenuCategoryButtons = props => (
-  props.restaurants.map(category => (
-    <button className="button-menu-header" type="button">{ category }</button>
-  ))
+  <div>
+    { props.restaurants.map(category => (
```
Comment (LittleJennie) [inline client/src/components/menuCategoryButtons.jsx]:
```suggestion
    { restaurants.map(category => (
```

File: client/src/components/menuEntry.jsx [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+  props.entries.map(entry => (
```
Comment (LittleJennie) [inline client/src/components/menuEntry.jsx]:
```suggestion
  entries.map(entry => (
```

File: client/src/components/menuEntry.jsx [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+    <div className="menu-item__2ZxJOnTY">
+      <div className="menu-price__1KiCimAr">{entry.menu_price}</div>
+      <div className="entry-item-name">{entry.menu_name}</div>
+      <div className="menu-description__2HXkC4oE">{entry.menu_description}</div>
+    </div>
+  ))
+);
+
```
Comment (LittleJennie) [inline client/src/components/menuEntry.jsx]: missing props validation

File: client/src/components/menusByHeaders.jsx [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+      <div className="menu-section-header__3nfLpHEA">
+        <h3 className="menu-section-title__22Q2IFWX">{menuHeader.header_name}</h3>
+      </div>
+      <div className="menu-items__2DRnPKGV">
+        <MenuEntry entries={menuHeader.entries} />
+      </div>
+    </div>
+  ))
+);
+
```
Comment (LittleJennie) [inline client/src/components/menusByHeaders.jsx]: missing props validation

File: client/src/components/menusByHeaders.jsx [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+export default MenuCategoryButtons;
```
Comment (LittleJennie) [inline client/src/components/menusByHeaders.jsx]: need new line

File: client/src/components/menusByHeaders.jsx [hunk 21]
```diff
@@ -0,0 +1,17 @@
...
+  props.menuHeaders.map(menuHeader => (
+    <div className="menu-section__14yfICNP">
+      <div className="menu-section-header__3nfLpHEA">
+        <h3 className="menu-section-title__22Q2IFWX">{menuHeader.header_name}</h3>
+      </div>
+      <div className="menu-items__2DRnPKGV">
+        <MenuEntry entries={menuHeader.entries} />
+      </div>
+    </div>
+  ))
```
Comment (LittleJennie) [inline client/src/components/menusByHeaders.jsx]: missing ;

File: client/src/components/menusByHeaders.jsx [hunk 22]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import MenuEntry from './MenuEntry.jsx';
+
+const MenuCategoryButtons = props => (
```
Comment (LittleJennie) [inline client/src/components/menusByHeaders.jsx]:
```suggestion
const MenuCategoryButtons =({ menuHeaders }) => (
```

File: client/src/components/menusByHeaders.jsx [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+  props.menuHeaders.map(menuHeader => (
```
Comment (LittleJennie) [inline client/src/components/menusByHeaders.jsx]:
```suggestion
  menuHeaders.map(menuHeader => (
```

Comment (cminho921) [reply client/src/components/menusByHeaders.jsx]: Will do code clean up by Eslint style guide

File: client/src/components/App.jsx (hunk 7 above)
Comment (cminho921) [reply client/src/components/App.jsx]: Thanks this looks much cleaner

File: client/src/components/menuCategoryButtons.jsx (hunk 12 above)
Comment (cminho921) [reply client/src/components/menuCategoryButtons.jsx]: this props destructure suggestion is really helpful

File: client/src/components/menuCategoryButtons.jsx (hunk 11 above)
Comment (cminho921) [reply client/src/components/menuCategoryButtons.jsx]: I will apply it!

**Pull request merged**